software-mansion/radon-ide
# Add recording controls enhancement

This PR introduces recording controls enhancement: 
- introduces a toggle button to start/stop recordings
- opens a save dialog immediately after pressing stop (in future, we could reuse replayOverlay for video previews and edits)
- ignores and does not save recordings under 1 second
- adds a maximum limit of 10 minutes for recordings, after which it cancels the recording
- change default save location to Desktop or homedir (on Windows) and add timestamp to filename
- eliminates text labels from the Replay and Logs buttons to create more space in the top button group
- updates the Replay button icon

### Test plan:
1. Start a recording, interact with the test app, then stop the recording after a few seconds and save the file to a selected location. Open the saved recording to verify that the app's screen was correctly captured.
2. Start a recording and stop it immediately to confirm that the save dialog does not appear for recordings under 1 second.
3. Test the maximum recording time to ensure that it automatically stops and opens the save dialog after the 5-minute limit is reached.
4. Ensure that the Replay button is disabled during recording.
5. After capturing a recording, test the Replay functionality to ensure it still works correctly.
6. Verify that the timestamps on the recorded video filenames reflect the correct date and time, matching the system clock.

Before:
![Screenshot 2024-11-15 at 17 22 48](https://github.com/user-attachments/assets/ffa8fb7c-a5f4-4e05-ade3-9ddd37390e1d)

After:
![Screenshot 2024-11-15 at 17 23 34](https://github.com/user-attachments/assets/ec215d2c-2d77-439c-aff0-3c963c22f994)
<img width="229" alt="Screenshot 2024-11-18 at 18 52 57" src="https://github.com/user-attachments/assets/6b803b88-b342-42e3-80ec-86b181a83694">
<video src="https://github.com/user-attachments/assets/03e4422f-829c-4b42-aa75-b8306ee96d91" />

resolves: #708
